pybamm/solvers/base_solver.py

```diff
@@ -775,6 +777,28 @@ def step(
         inputs = inputs or {}
         ext_and_inputs = {**external_variables, **inputs}
 
+        # Check that any inputs that may affect the scaling have not changed
+        # Set model timescale
+        temp_timescale_eval = model.timescale.evaluate(inputs=inputs)
+        # Set model lengthscales
+        temp_length_scales_eval = {
+            domain: scale.evaluate(inputs=inputs)
+            for domain, scale in model.length_scales.items()
+        }
+        if old_solution is not None:
+            if temp_timescale_eval != old_solution.timescale_eval:
+                raise pybamm.SolverError(
+                    "The model timescale is a function of an input parameter "
+                    "and the value has changed between steps!"
+                )
+            for domain in temp_length_scales_eval.keys():
+                old_dom_eval = old_solution.length_scales_eval[domain]
+                if temp_length_scales_eval[domain] != old_dom_eval:
+                    pybamm.logger.error(
+                        "The model lengthscale is a function of an input "
+                        "parameter {} and the value has changed between "
+                        "steps!".format(domain)
```

Comment:
Check this string ("domain" is not an input parameter). Should it say "an input parameter affecting the length scale in {}"?

Reply:
Re-worded